agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/context/AIAgentLLMContext.kt

```diff
@@ -36,6 +36,26 @@ public data class AIAgentLLMContext(
     private val clock: Clock
 ) {
 
+    /**
+     * Creates a deep copy of this LLM context.
+     *
+     * @return A new instance of [AIAgentLLMContext] with deep copies of mutable properties.
+     */
+    public suspend fun copy(): AIAgentLLMContext {
+        return rwLock.withReadLock {
+            AIAgentLLMContext(
+                tools.toList(),
+                toolRegistry,
+                prompt.copy(),
```

Comment:
There's no need to unnecessary copy prompt and tools, these are already fully immutable